p2panda-net/src/addrs.rs

```diff
@@ -46,18 +48,48 @@ impl From<RelayUrl> for IrohRelayUrl {
     }
 }
 
-/// A node address including public key, socket address(es) and an optional relay URL.
-pub type NodeAddress = (PublicKey, Vec<SocketAddr>, Option<RelayUrl>);
+/// Converts a `iroh` relay url type to the `p2panda-net` implementation.
+pub(crate) fn to_relay_url(url: IrohRelayUrl) -> RelayUrl {
+    RelayUrl(url)
+}
+
+/// Node address including public key, socket address(es) and an optional relay URL.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NodeAddress {
+    pub public_key: PublicKey,
+    pub direct_addresses: Vec<SocketAddr>,
+    pub relay_url: Option<RelayUrl>,
+}
+
+impl NodeAddress {
+    pub fn from_public_key(public_key: PublicKey) -> Self {
```

Comment:
Did you consider rather implementing `From`? Same for `NodeAddress` / `IrohNodeAddr`.

Just asking out of curiosity; I think this is fine as it is.